### DISCUSSION TITLE
Create 1D Basis with spin_basis_general (need multithreading feature)

### DISCUSSION THREAD

**Discussion by amoyerjlab** [QuSpin/QuSpin#470] [Q&A]
I have a script to study the 1D Haldane Model. However, the function spin_basis_1D() doesn't support multithreading. I need multithreading since my memory use is exceeding our computer's available hardware. 
1) Is it possible to build a 1D model with spin_basis_general() function?
2) If it is, can you explain how I can implement these additional symmetry arguments in spin_basis_general:

basis_ground=spin_basis_1d(L, pauli=False, S="1", Nup = L, kblock=L, pblock=1, zblock=1)
(This set of symmetries contains the ground state and cuts down on resources)

I am trying to follow the example here: https://weinbe58.github.io/QuSpin/generated/quspin.basis.spin_basis_general.html
In the example script, they have this line:

basis_2d = spin_basis_general(N_2d,kxblock=(T_x,0),kyblock=(T_y,0),pxblock=(P_x,0),pyblock=(P_y,0),zblock=(Z,0))
Unlike the 1d function, the symmetry arguments now take a tuple of a list and 0. But I don't know how to create the list for my 1D model. Or if I even can.

I am just a postbac level researcher. I haven't formally taken a CMP class, so my physics knowledge is mostly from the Articles I have read.

**Comment by weinbe58** [comment]
Hi! 

You almost have it, In your code the 1D symmetries can be generated by setting Ly=1. In this case you'll see that the transformation T_y and P_y do not do anything anymore so you can just remove those. Just note that when kblock != 0 or L/2 you can't use the parity symmetry. This shouldn't be an issue if you are only working in the ground state sector.
